File: 2ºCARRERA/tprog/practica2_872815_873338/java/Over.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Over.java
 */

import java.util.Stack;

public class Over implements Instruccion {
    public int ejecuta(Stack<Integer> pila, int pc) {
        int num = pila.pop();
        int num2 = pila.peek();
        pila.push(num);
        pila.push(num2);
        pc++;
        return pc;
    }

    public void muestra() {
        System.out.print("over");
    }
}

File: 2ºCARRERA/tprog/practica2_872815_873338/c++/over.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : over.cc
 */

#include "over.h"

void Over::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num = pila.top();
    pila.pop();
    int32_t num2 = pila.top();
    pila.push(num);
    pila.push(num2);
    pc++;
}

void Over::muestra(){
    cout << "over";
}